src/utils/jsDocUtils.ts

```diff
@@ -17,16 +18,21 @@ export function getJSDocComment(node: ts.Node, tagName: string) {
   return tags[0].comment;
 }
 
-export function getJSDocTagNames(node: ts.Node) {
+export function getJSDocTagNames(node: ts.Node, requireTagName = false) {
   let tags: ts.JSDocTag[];
   if (node.kind === ts.SyntaxKind.Parameter) {
     const parameterName = ((node as any).name as ts.Identifier).text;
     tags = getJSDocTags(node.parent as any, tag => {
-      return tag.comment !== undefined && tag.comment.startsWith(parameterName);
+      if (ts.isJSDocParameterTag(tag)) {
+        return false;
+      } else if (tag.comment === undefined) {
+        throw new GenerateMetadataError(`Orphan tag: @${tag.tagName.text || tag.tagName.escapedText} should have a parameter name follows with.`);
+      }
+      return tag.comment.startsWith(parameterName);
     });
   } else {
     tags = getJSDocTags(node as any, tag => {
-      return tag.comment !== undefined;
+      return requireTagName ? tag.comment !== undefined : true;
```

Comment:
```suggestion
      return requireTagName ? true:  tag.comment !== undefined;
```

Reply:
I think `requireTagName` represents the match function in `getJSDocTags` should check `tag.comment` is `undefined` or not.  
  
Am I right?

Reply:
The match function was `tag => tag.comment !== undefined;`. Now it's `tag => requireTagName ? tag.comment !== undefined : true;`

I guess I just misinterpreted the meaning of the parameter since it, as an API consumer would likely mean "do I need this thing to have a comment" (since that's the behavioral difference). So what I thought is (reading the parameter name) the default for "do I need this thing to have a comment" should be true.

It's not a big difference either way, I'll merge this before the next release anyway.

Reply:
So I don't need further changes right?

If yes, please mark this resolve, thanks you. 😃